File: POPBackend/src/main/java/com/ur/popbackend/entities/Member.java
package com.ur.popbackend.entities;

import com.ur.popbackend.enums.Availability;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.time.LocalDate;
import java.util.TimeZone;

@Entity
public class Member {
    @Id
    @GeneratedValue
    private int id;
    private String name;
    private LocalDate companyAnniversary;
    private TimeZone timeZone;
    private Availability availability;
    private String phone;
    private LocalDate startingDate;
    private String companyRole;
    private String avatarURI;
    private String email;
    private String bio;

    public Member() {
        //spring data
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public LocalDate getCompanyAnniversary() {
        return companyAnniversary;
    }

    public void setCompanyAnniversary(LocalDate companyAnniversary) {
        this.companyAnniversary = companyAnniversary;
    }

    public TimeZone getTimeZone() {
        return timeZone;
    }

    public void setTimeZone(TimeZone timeZone) {
        this.timeZone = timeZone;
    }

    public Availability getAvailability() {
        return availability;
    }

    public void setAvailability(Availability availability) {
        this.availability = availability;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public LocalDate getStartingDate() {
        return startingDate;
    }

    public void setStartingDate(LocalDate startingDate) {
        this.startingDate = startingDate;
    }

    public String getCompanyRole() {
        return companyRole;
    }

    public void setCompanyRole(String companyRole) {
        this.companyRole = companyRole;
    }

    public String getAvatarURI() {
        return avatarURI;
    }

    public void setAvatarURI(String avatarURI) {
        this.avatarURI = avatarURI;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getBio() {
        return bio;
    }

    public void setBio(String bio) {
        this.bio = bio;
    }
}

File: POPMobile/app/src/main/java/com/ur/popandroid/entities/Member.kt
package com.ur.popandroid.entities

import android.os.Parcel
import android.os.Parcelable


data class Member(
    val id: Int,
    val name: String,
    val companyAnniversary: String,
    val timeZone: String,
    val availability: String,
    val phone: String,
    val startingDate:String,
    val companyRole:String,
    val avatarURI: String,
    val email: String,
    val bio: String
    ) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readInt(),
        parcel.readString() !!,
        parcel.readString() !!,
        parcel.readString() !!,
        parcel.readString() !!,
        parcel.readString() !!,
        parcel.readString() !!,
        parcel.readString() !!,
        parcel.readString() !!,
        parcel.readString() !!,
        parcel.readString() !!
    )
    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(id)
        parcel.writeString(name)
        parcel.writeString(companyAnniversary)
        parcel.writeString(timeZone)
        parcel.writeString(availability)
        parcel.writeString(phone)
        parcel.writeString(startingDate)
        parcel.writeString(companyRole)
        parcel.writeString(avatarURI)
        parcel.writeString(email)
        parcel.writeString(bio)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Member> {
        override fun createFromParcel(parcel: Parcel): Member {
            return Member(parcel)
        }

        override fun newArray(size: Int): Array<Member?> {
            return arrayOfNulls(size)
        }
    }
}